Civi/Webshoptools/HandleWebShopActivity.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Civi\Webshoptools;
+
+class HandleWebShopActivity {
+
+  /**
+   * @param $operation
+   * @param $objectName
+   * @param $id
+   * @param $params
+   * @return void
+   */
+  public static function run($operation, $objectName, $id, &$params) {
+    if (empty($params['custom']) || !($objectName == 'Activity' && ($operation == 'create' || $operation == 'edit'))) {
+      return;
+    }
+
+    $currentActivity = $operation == 'edit' ? civicrm_api3('Activity', 'getsingle', ['id' => $params['id']]) : null;
+    $activityTypeId = $operation == 'edit' ? $currentActivity['activity_type_id'] : $params['activity_type_id'];
+
+    if (self::getWebShopActivityTypeId() != $activityTypeId) {
+      return;
+    }
+
+    $customFieldsData = self::getWebShopCustomFieldsData($params, $currentActivity);
+    $params['subject'] = self::generateSubject($customFieldsData);
+
+    if (self::isNewActivityStatusCompleted($currentActivity, $params, $operation)) {
+      self::setCustomValue('order_exported', '1', $customFieldsData, $params);
+      self::setCustomValue('order_exported_date', \CRM_Utils_Date::currentDBDate(), $customFieldsData, $params);
```

Comment:
For `order_exported_date`, we discussed this internally and decided we only want to set the date if a) the current value is empty and b) no new value is provided. Basically: If the user/API call manually sets a date (or has done so in the past), that date should win over `currentDBDate()`.